src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs

```diff
@@ -3392,5 +3392,29 @@ private RestoreCommandProvidersCache GetRestoreProviderCache()
 
             return _restoreProviderCache;
         }
+
+        public static async Task<HashSet<string>> GetTargetFramework(NuGetProject project)
+        {
+            var frameworks = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+            NuGetFramework framework;
+            if (project.TryGetMetadata<NuGetFramework>(NuGetProjectMetadataKeys.TargetFramework, out framework))
+            {
+                frameworks.Add(framework.ToString());
+            }
+
+            if (project is BuildIntegratedNuGetProject)
+            {
+                var biproject = project as BuildIntegratedNuGetProject;
+                var dgcContext = new DependencyGraphCacheContext();
+                var packageSpecs = await biproject.GetPackageSpecsAsync(dgcContext);
+                foreach (var packageSpec in packageSpecs)
+                {
+                    frameworks = frameworks.Concat(packageSpec.TargetFrameworks.Select(tf => tf.FrameworkName.ToString())).ToHashSet();
```

Comment:
Why do we need a hash set? 

We can probably add an interface method to get the frameworks for Build Integrated projects.

cc @zivkan thoughts?

Reply:
I chose to use a HashSet because I want a collection of the unique frameworks used without duplicates. I'm happy to change it to some other type if it's more appropriate. What do you suggest?

Reply:
sorry, didn't see this mention for an entire week :(

Regarding hashset vs list, it's complicated :)  if it's a performance question, we'd really need to do a microbenchmark to be sure. `HashSet` has overheads, so I've often read that when the collection is small, it's faster to use `List`. However, semantically how we're using it, to find a unique collection of TFMs, `HashSet` is the perfect fit conceptually. Having said that, adding new items to the `HashSet` by using `.Concat`, `.Select` and `.ToHashSet` is just about the worst performance without intentionally writing slow code. If HashSet doesn't have a `AddRange`, iterate `packageSpec.TargetFrameworks` and `.Add` one at a time. That's what `.ToHashSet` does internally.

I don't fully understand the comment about adding an interface method for build integrated projects. Are you suggesting moving this method to `BuildIntegratedNuGetProject`?  If we plan to support package recommendations for `PackageReference` projects only, then that makes sense.

Reply:
> I don't fully understand the comment about adding an interface method for build integrated projects. Are you suggesting moving this method to BuildIntegratedNuGetProject? If we plan to support package recommendations for PackageReference projects only, then that makes sense.

There is a way to get the target framework in PC projects already. 
What makes build integrated projects special is the fact that they can multi target.

Reply:
Is there an additional change I should make here? Or is this okay now?

Reply:
This method is now complete removed, so I imagine it's ok (I did not look, but inferred from the other comments).